File: app/models/challenge_response.py
from pydantic import BaseModel, Field
from datetime import datetime

class ChallengeResponse(BaseModel):
    challenge_id: str = Field(..., description="Unique identifier for the challenge")
    challenge: str = Field(..., description="Challenge string to be signed by the client")
    expires_at: datetime = Field(..., description="UTC timestamp when the challenge expires")

    class Config:
        schema_extra = {
            "example": {
                "challenge_id": "83fcfcf6e2e84df7b7a84db6c52934f7",
                "challenge": "e9f34c6d9c0b4f74a1f9f3a2e5a1b3c4",
                "expires_at": "2025-07-31T23:59:59Z"
            }
        }

File: apps/cli/lib/src/models/challenge_response.dart
class ChallengeResponse {
  ChallengeResponse({
    required this.challengeId,
    required this.challenge,
    required this.expiresAt,
    required this.challengeCode,
  });

  factory ChallengeResponse.fromJson(Map<String, dynamic> json) {
    return ChallengeResponse(
      challengeId: json['challenge_id'] as String,
      challenge: json['challenge'] as String,
      expiresAt: DateTime.parse(json['expires_at'] as String),
      challengeCode: json['challenge_code'] as String,
    );
  }

  final String challengeId;
  final String challenge;
  final DateTime expiresAt;
  final String challengeCode;
}
